server/parser.go

```diff
@@ -1175,3 +1180,18 @@ func (c *client) clonePubArg() error {
 		}
 	}
 }
+
+func (ps *parseState) getHeader() http.Header {
+	if ps.header == nil {
+		hdr := ps.pa.hdr
```

Comment:
you could do one line:
```
if hdr := ps.pa.hdr; hdr > 0 {
  ..
```
Should you move to a `func (c *client) getHeader()`, it would be `c.pa.hdr`. And again, `c.pa.header` if you were to move `header` to inside `pa` (`pubArg`).